### ISSUE TITLE
Loss NaN with existing PyTorch Implementation

### ISSUE THREAD

**Issue by ZhouTimeMachine** [Gsunshine/meanflow#4]
Hello! I try to run the full training process of meanflow, and I test 2 popular PyTorch-based repo:

- [MeanFlow - haidog-yaqub](https://github.com/haidog-yaqub/MeanFlow/blob/main/train.py)
- [MeanFlow-PyTorch - HaoyiZhu](https://github.com/HaoyiZhu/MeanFlow-PyTorch)

Both will encounter the loss nan problem.
 
- the former try to train 200k steps by default, nan comes between 150k - 200k steps
- the latter try to train 400k steps by default, nan comes between 350k - 400k steps

Apology that I'm not familar with jax, so I haven't run this repo directly. What might be the cause of the loss nan problem? Is this problem present in this repository implemented by jax? Thanks for any experiences or suggestions.

**Comment by stefenmax** [comment]
Hi, I also run into similar issue when run using the haidog version. By changed the https://github.com/haidog-yaqub/MeanFlow/blob/a2c1e5c2ae0bacdb4c9eeedb000f0e9a053ecf1d/train.py#L19C5-L19C54 to 'bf16'. I didn't get nan when trained with cifar10 in default setting.

**Comment by ZhouTimeMachine** [comment]
> Hi, I also run into similar issue when run using the haidog version. By changed the [haidog-yaqub/MeanFlow@`a2c1e5c`/train.py#L19C5-L19C54](https://github.com/haidog-yaqub/MeanFlow/blob/a2c1e5c2ae0bacdb4c9eeedb000f0e9a053ecf1d/train.py#L19C5-L19C54) to 'bf16'. I didn't get nan when trained with cifar10 in default setting.

Thanks! I found it really works for haidog-yagub's version, and I believe it may also work for Haoyi Zhu's version.
